lib/hamster/associable.rb

```diff
@@ -45,5 +45,28 @@ def update_in(*key_path, &block)
       end
       put(key, new_value)
     end
+
+    # Return the value of successively indexing into a container
+    # If any of the keys is not present in the container, return `nil`.
+    # keys that the Hamster type doesn't understand, raises an argument error
+    #
+    # @example
+    #   h = Hamster::Hash[:a => 9, :b => Hamster::Hash[:c => 'a', :d => 4], :e => nil]
```

Comment:
I would give an example that has a Hash+Vector together, like the first example of `#update_in` above. This shows that both containers can be used together with the same method.